core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -167,12 +167,24 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
       StreamExecutionEnvironment.getExecutionEnvironment
     }
 
-    if (!env.getCheckpointConfig.isCheckpointingEnabled()) {
+    if (!env.getCheckpointConfig.isCheckpointingEnabled() && isDefaultCheckpointingEnabled(config, streamlet)) {
       setDefaultCheckpointing(env)
     }
     env
   }
 
+  /**
+   * This checks whether the user, through configuration, has disabled checkpointing
+   * by setting flink.execution.checkpointing.interval value to less then 0
+  **/
+  def isDefaultCheckpointingEnabled(config: Config, streamlet: String): Boolean = {
+    val runtimePath   = "cloudflow.runtimes.flink.config.cloudflow.checkpointing.default"
+    val streamletPath = s"cloudflow.streamlet.${streamlet}.config.cloudflow.checkpointing.default"
+    if (config.hasPath(runtimePath) && config.getBoolean(runtimePath) == false) return false
+    if (config.hasPath(streamletPath) && config.getBoolean(streamletPath) == false) return false
```

Comment:
I think this condition doesn't exactly cover the use case e.g.:
- "cloudflow.runtimes.flink.config.cloudflow.checkpointing.default" = false
- "cloudflow.streamlet.my-streamlet.config.cloudflow.checkpointing.default" = true
will return `false` for the streamlet `my-streamlet` instead of `true` (e.g. the streamlet configuration has to be "more specific" and override the runtime config)

Reply:
great catch! I'll fix it.

Reply:
minor suggestion here, since you are at it, can I ask if you can use a single `if - else if - else` structure (or even a `match case`) instead of using early returns? It will make the code easier to reason about and more "scala-idiomatic" 🙂

Reply:
added "find checkpointing is disabled for streamlet overriding runtimes being enabled"